Source/WTF/wtf/Assertions.h

```diff
@@ -484,6 +520,15 @@ WTF_EXPORT_PRIVATE NO_RETURN_DUE_TO_CRASH void WTFCrashWithSecurityImplication(v
 
 #if LOG_DISABLED
 #define LOG_WITH_LEVEL(channel, level, ...) ((void)0)
+#elif USE(RDK_LOGGER)
+#define LOG_WITH_LEVEL_STRING(channel, level, ...) do { \
+    rdk_LogLevel levelMap[] = { RDK_LOG_NOTICE, RDK_LOG_ERROR, RDK_LOG_WARN, RDK_LOG_INFO, RDK_LOG_DEBUG}; \
```

Comment:
rdk_LogLevel levelMap should probably be statically defined in one place rather than inlined in each define instance.